File: Rust/1431 Kids With the Greatest Number of Candies/Solution.rs
// URL: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/
// Run result: Runtime: 0 ms, faster than 100.00% of Rust online submissions for Kids With the Greatest Number of Candies.
//             Memory Usage: 2 MB, less than 100.00% of Rust online submissions for Kids With the Greatest Number of Candies.
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let mut ret: Vec<bool> = Vec::new();
        let max_candies = candies.iter().max().unwrap();
        for v in candies.iter() {
            if v + extra_candies >= *max_candies {
                ret.push(true);
            } else {
                ret.push(false);
            }
        }
        return ret
    }
}

File: Python/1431 Kids With the Greatest Number of Candies/Solution.py
# URL: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/
# Run result: Runtime: 56 ms, faster than 33.33% of Python3 online submissions for Kids With the Greatest Number of Candies.
#             Memory Usage: 13.9 MB, less than 100.00% of Python3 online submissions for Kids With the Greatest Number of Candies.
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        ret_list = [False] * len(candies)
        max_candies = max(candies)
        for i in range(len(candies)):
            if candies[i] + extraCandies >= max_candies:
                ret_list[i] = True
        return ret_list